packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -103,12 +104,17 @@ export class XInstaller {
    * Receives the {@link InstallXOptions} and merges it with the default fallback options. Also
    * creates the public {@link XAPI}.
    *
+   * @remarks Auto initializes the Vue application if window.initX is defined as a function or
+   * object specifying the {@link SnippetConfig | snippet config}.
+   *
+   *
    * @param options - The {@link InstallXOptions}.
    *
    * @public
    */
   public constructor(protected readonly options: InstallXOptions) {
     this.createAPI();
+    this.autoInit();
```

Comment:
I like this suggestion. To me feels strange that just creating the installer also initialises the app out of the box.  However, wouldn't be better the following?:

1. Refactor `autoInit` into something like `getSnippetConfig(): SnippetConfig` .
2.  Refactor `init` to call `getSnippetConfig` first of all.
3. Just call `new XInstaller(options).init()`.

Reply:
Even though I like your approach @diegopf , I think we should still support the option of passing a `SnippetConfig` as a parameter to the `XInstaller.init` function, as `x-components` can be used as a `library`, hence there shouldn't be a need of injecting the `snippet config` object into the `window` object.

Reply:
@diegopf The point is that there are 2 ways that the XInstaller can be initialised:

1. There is a defined `initX` object in `window` at the time the x-components are downloaded
  a. It is an static object of `SnippetConfig` type.
  b. It is a function that should return a `SnippetConfig` object.
2. There is no `initX` object defined. Therefore, the developer that is integrating the XComponents into his website should call `window.X.init(snippetConfig)` manually when he is ready, which will call `XInstaller.init` method.

With your suggestion I'm not sure how this last requirement fits. If we didn't need to support it I would like it. Maybe I should have suggested to rename `autoInit` to `tryAutoInit` which describes better that it might not actually initialise the project?.

In the case that @LuisMartinez15 says, using XComponents as a standard components library in your Vue project, we shouldn't use the `XInstaller`at all, that's why I'm willing to move it together with other archetype utilities into its own project, so there is no confusion with that.

Reply:
I've refactored it based on your suggestions. Let me know if you find it okey or want to suggest further changes.